### PR TITLE
exclude non-latest versions of docs from broken link checker

### PR DESCRIPTION
**Pull request by fjlanasa**
### Pull Request Description
Follow up to https://github.com/alteryx/evalml/pull/3398

The crawler ended up following a link on the latest version of the docs to the stable version of the docs, which did not contain a fix to a broken link, leading to a failed check. This PR updates the `exclude` argument to not check documentation pages for versions other than latest. 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: .github/workflows/broken_link_checker.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
         uses: ruzickap/action-my-broken-link-checker@v2
         with:
           url: https://evalml.alteryx.com/en/latest/index.html
-          cmd_params: '--max-connections=10 --color=always --ignore-fragments --buffer-size=8192 --exclude="(twitter|github|cloudflare)"'
+          cmd_params: '--max-connections=10 --color=always --ignore-fragments --buffer-size=8192 --exclude="(twitter|github|cloudflare|evalml\\.alteryx\\.com\\/en\\/(stable|main|v.+).*)"'
```
Comment (freddyaboulton) [inline .github/workflows/broken_link_checker.yml:18]: Nice! Did you run this to make sure it would pass?

Comment (fjlanasa) [reply .github/workflows/broken_link_checker.yml:18]: Yep, ran it locally with the `muffet` cli and also edited the action to run on push so i could test it in github. https://github.com/alteryx/evalml/runs/5678469196?check_suite_focus=true

Comment (freddyaboulton) [reply .github/workflows/broken_link_checker.yml:18]: Nice - need to look into muffet cli

**Review by freddyaboulton — APPROVED** [review]
Thank you for the fix @fjlanasa !

**Pull request merged**